packages/blade/src/components/ActionList/ActionListItem.tsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import styled from 'styled-components';
+import { StyledActionListItem } from './StyledActionListItem';
+import { componentIds } from './componentIds';
+import type { StyledActionListItemProps } from './getBaseActionListItemStyles';
+import Box from '~components/Box';
+import type { IconComponent } from '~components/Icons';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { Text } from '~components/Typography';
+import { isReactNative, makeAccessible, makeSize } from '~utils';
+import { Checkbox } from '~components/Checkbox';
+
+const ActionListItemContext = React.createContext<{
+  intent?: ActionListItemProps['intent'];
+}>({});
+
+const SectionDivider = styled(Box)((props) => ({
+  // @TODO: replace this with token value if we add 1px token
+  height: makeSize(1),
+  backgroundColor: props.theme.colors.surface.border.normal.lowContrast,
+  margin: `${makeSize(props.theme.spacing[1])} ${makeSize(props.theme.spacing[3])}`,
+}));
+
+const StyledActionListSectionTitle = styled(Box)((props) => ({
+  padding: makeSize(props.theme.spacing[3]),
+}));
+
+const ActionListSection = ({
+  title,
+  children,
+  hideDivider,
+}: {
+  title: string;
+  children: React.ReactNode[] | React.ReactNode;
+  hideDivider?: boolean;
```

Comment:
nope. renamed it to `_hideDivider` and marked `@private` in jsdoc.

Also removed this entire logic from web and relying on descendant styling of css there.

Keeping it in react native because there isnt any other simpler way.

Reply:
Oh more context: I am looping over children and checking if ActionListSection is the final section and  we hide the divider there.

https://github.com/razorpay/blade/blob/f32c66695951d71312aa96c27485cf5c3e072cd2/packages/blade/src/components/ActionList/actionListUtils.ts#L119-L120